src/VSCodeUI.ts

```diff
@@ -53,6 +54,47 @@ export namespace VSCodeUI {
         }
     }
 
+    export function setupEnvironment(terminal: Terminal): void {
+        // do this first so it can be overridden if desired
+        setJavaHomeIfAvailable(terminal);
```

Comment:
According to the call sequence of `setJavaHomeIfAvailable` and `environmentSettings.forEach`, if I set both `java.home` and `maven.terminal.customEnv[JAVA_HOME]`, the latter overrides the former. Is it the expected behaviour for most users? Or how to avoid ambiguity?

Reply:
IMHO, that is expected behavior.  If `java.home` is configured, then we can reuse it.  But if a user also configures `JAVA_HOME` for our extension, then it should take precedence.

I will explain that in the README.